### PR TITLE
Mention

### PR DESCRIPTION
**Pull request by Eschults**
implemented react-mentions suggestions in jobs#new, questions#new, milestones#new, answers#edit, answers#new
implemented slack notif in PM to user mentioned in answers#edit, answers#new
ping @ssaunier

### CODE REVIEW

File: app/assets/javascripts/components/gh_nickname_mention.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    } else {
+      AnswerActions.update(this.props.id, value)
+    }
+  },
+
+  displayTransform: function(id, display) {
+    return '@' + id;
+  },
+
+  handleRemove: function() {
```
Comment (ssaunier) [inline app/assets/javascripts/components/gh_nickname_mention.jsx]: Do we need those 2 methods? If they just log, we can remove them (also remove line 44 and 45)

File: app/assets/javascripts/components/language_detection_textarea.jsx [hunk 2]
```diff
@@ -43,27 +43,26 @@ class LanguageDetectionTextarea extends React.Component {
...
-          className='answer-form-input answer-form-input-ui'
-          onKeyUp={this.onKeyUp.bind(this)}
-          onKeyDown={this.props.onKeyDown}
-        />
+        <GhNicknameMention {...this.props} ref='content' onKeyUp={this.onKeyUp.bind(this)} />
         <div className='answer-form-preview' dangerouslySetInnerHTML={{__html: this.state.renderedContent}}></div>
       </div>
     )
   }
+  // <textarea
```
Comment (ssaunier) [inline app/assets/javascripts/components/language_detection_textarea.jsx]: Dead code => remove

File: app/views/jobs/_job.json.jbuilder [hunk 3]
```diff
@@ -3,5 +3,9 @@ json.city job.city
 json.company job.company
 json.contact_email job.contact_email
 json.ad_url job.ad_url
-json.content render_markdown(job.description)
-json.original_content job.description
+enriched_content = job.description.dup
+job.description.scan(/\B@\S*/).flatten.uniq.each do |mention|
```
Comment (ssaunier) [inline app/views/jobs/_job.json.jbuilder]: Duplicated code. Should go to a helper.

File: app/views/questions/_question.json.jbuilder [hunk 4]
```diff
@@ -1,4 +1,8 @@
 json.partial! "posts/post", post: question
-json.content render_markdown(question.content)
-json.original_content question.content
+enriched_content = question.content.dup
+question.content.scan(/\B@\S*/).flatten.uniq.each do |mention|
```
Comment (ssaunier) [inline app/views/questions/_question.json.jbuilder]: Same here.

**Comment by ssaunier** [comment]
Could you create an issue in this repo and list all the changes done in the `react-mentions.js` library?

**Comment by Eschults** [comment]
done @ssaunier

**Pull request merged**